targets/targets.json

```diff
@@ -5306,6 +5306,27 @@
             "target.network-default-interface-type": "ETHERNET"
         }
     },
+    "S5JS100": {
+        "inherits": ["Target"],
+        "core": "Cortex-M7",
+        "components_add": ["FLASHIAP"],
+        "supported_toolchains": ["GCC_ARM", "IAR", "ARM"],
```

Comment:
Just because we had some recent changes regarding what `"ARM"` means in this field as of 5.12.0, does this target support ARMC5 and ARMC6?

Reply:
Hi,
As you know, all the recent Mbed-OS master based code compiles only in ARMC6. I couldn't test in ARMC5 because toolchain forced ARMC6 compilation. Here is the ARMC6 compile log: [build_log_S5JS100_ARM.txt](https://github.com/ARMmbed/mbed-os-rjio/files/3016168/build_log_S5JS100_ARM.txt)
```
/home/andcho01/tools/ARMCompiler6.12/bin/armclang -c --target=arm-arm-none-eabi -mthumb -Os -Wno-armcc-pragma-push-pop -Wno-armcc-pragma-anon-unions -DMULADDC_CANNOT_USE_R7 -fdata-sections -fno-exceptions -MMD -D_LIBCPP_EXTERN_TEMPLATE(...)= -fshort-enums -fshort-wchar -DMBED_TRAP_ERRORS_ENABLED=1 -mcpu=cortex-m7 -fno-rtti -std=gnu++98 -DMBED_MPU_CUSTOM -D__MBED__=1 -DTARGET_LIKE_MBED -DMBED_BUILD_TIMESTAMP=1553745562.35 -DDEVICE_PORTINOUT=1 -D__MBED_CMSIS_RTOS_CM -DCOMPONENT_PSA_SRV_EMUL=1 -D__CMSIS_RTOS -DTOOLCHAIN_ARMC6 -DARM_MATH_CM7 -DTARGET_CORTEX_M -DTARGET_LIKE_CORTEX_M7 -DTARGET_S5JS100 -DTARGET_M7 -DCOMPONENT_PSA_SRV_IMPL=1 -DCOMPONENT_FLASHIAP=1 -DCMSDK_CM7 -DDEVICE_INTERRUPTIN=1 -DTARGET_CORTEX -DDEVICE_PORTOUT=1 -D__CORTEX_M7 -DTARGET_Samsung -DDEVICE_PORTIN=1 -DTARGET_RELEASE -DCOMPONENT_NSPE=1 -DMBED_TEST_MODE -DTARGET_SIDK_S5JS100 -DDEVICE_TRNG=1 -DTARGET_RTOS_M4_M7 -DDEVICE_SLEEP=1 -DDEVICE_SPI=1 -DTARGET_NAME=S5JS100 -DDEVICE_SERIAL=1 -DDEVICE_FLASH=1 -DTOOLCHAIN_ARM -DTOOLCHAIN_ARM_STD @./BUILD/tests/S5JS100/ARM/.includes_700d7631b8f26d37481060463f56c96d.txt -include ./BUILD/tests/S5JS100/ARM/mbed_config.h -o BUILD/tests/S5JS100/ARM/mbed-os/features/netsocket/InternetSocket.o ./mbed-os/features/netsocket/InternetSocket.cpp
```

Also please note that I upgraded IAR as well to 8.32.2 as the latest Mbed-OS suggests.
[build_log_S5JS100_IAR.txt](https://github.com/ARMmbed/mbed-os-rjio/files/3016167/build_log_S5JS100_IAR.txt)